### PR TITLE
[WIP] Web Upload

### PR DESCRIPTION
**Pull request by kngu9**
This is currently a WIP.

### CODE REVIEW

File: internal/api/thumbnails.go [hunk 1]
```diff
@@ -2,10 +2,12 @@ package api
 
 import (
 	"fmt"
-	"github.com/gin-gonic/gin"
-	"github.com/photoprism/photoprism/internal/photoprism"
 	"log"
 	"strconv"
+
```
Comment (lastzero) [inline internal/api/thumbnails.go]: Please run "make fmt" before pushing, it should order the imports alphabetically

File: internal/api/photos.go [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+	filetype "gopkg.in/h2non/filetype.v1"
+
 	"github.com/photoprism/photoprism/internal/forms"
 	"github.com/photoprism/photoprism/internal/photoprism"
-	"net/http"
-	"strconv"
 )
 
+type PhotosResponse {
+	message: interface{}
```
[the remark this answers is not in the document]
Comment (kngu9) [reply internal/api/photos.go]: I was planning to normalize the output of the results. This is meant to be a generic response for the photo get and post handler.

File: internal/api/photos.go [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+	"github.com/google/uuid"
+	filetype "gopkg.in/h2non/filetype.v1"
+
 	"github.com/photoprism/photoprism/internal/forms"
 	"github.com/photoprism/photoprism/internal/photoprism"
-	"net/http"
-	"strconv"
 )
 
+type PhotosResponse {
```
Comment (Shivakishore14) [inline internal/api/photos.go]: Does this compile ? Looks like its missing `struct` keyword.

File: internal/api/photos.go (hunk 2 above)
Comment (Shivakishore14) [inline internal/api/photos.go]: Why is there a `:` after `message`. is this a typo or am i missing something here?.

File: frontend/package.json [hunk 4]
```diff
@@ -93,5 +94,9 @@
     "> 1%",
     "last 2 versions",
     "not ie <= 8"
-  ]
+  ],
+  "devDependencies": {
+    "@babel/core": "^7.1.2",
```
Comment (lastzero) [inline frontend/package.json]: Don't use devDependencies - everything is dev (or not) since we build the final JS file :)

**Comment by lastzero** [comment]
@kngu9 Are you still working on this? We are not in a hurry, just asking. We can also chat if you have questions or would like to get to know us.

**Comment by kngu9** [comment]
Hi @lastzero, I am however you are free to take over the feature if you'd like. Currently been boggled down with work :(

**Comment by lastzero** [comment]
Alright, reach out if you need anything. I'm totally busy with infrastructure and [documentation](https://github.com/photoprism/photoprism/wiki) at the moment. I also want to create proper components, right now half of the frontend is in `photos.vue`.

**Comment by kngu9** [comment]
@lastzero I'll try to work on it after the holidays, if it's still available. I'll close this issue, so people won't take the assumption that this is still being worked on.

**Pull request closed without merging**